airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -654,10 +637,30 @@ definitions:
               - GET
               - POST
         default: GET
-      request_options_provider:
+      request_body_data:
         anyOf:
-          - "$ref": "#/definitions/CustomRequestOptionsProvider"
-          - "$ref": "#/definitions/InterpolatedRequestOptionsProvider"
+          - type: string
+          - type: object
+            additionalProperties:
+              type: string
+      request_body_json:
+        anyOf:
+          - type: string
```

Comment:
Since these are a complex object that we reuse a little bit, might be worth defining a new property called RequestInput. I think we did this for PrimaryKey as some prior art

Reply:
If I understand your point properly, I'm a little bit reluctant to do that. My reasoning is that I feel like those fields are not representing the same thing and therefore might not evolve in the same way. One example I see of this is https://github.com/airbytehq/airbyte/issues/20734 where `request_body_json` would have a very different type definition than the others. I think we can argue the same for the other fields. Therefore, I feel like re-using a complex type for the different `request_*` field might lead to more work later when these will evolve differently. 

Let me know if this makes sense for you. If not, just say it and I'll define the `RequestInput` as suggested

Reply:
Fair enough, I agree with your rationale and we can leave as is! Thanks for adding the descriptions too!